exporter/zipkinexporter/zipkin.go

```diff
@@ -175,6 +212,49 @@ func (ze *zipkinExporter) Shutdown() error {
 	return ze.reporter.Close()
 }
 
+func (ze *zipkinExporter) PushTraceData(ctx context.Context, td consumerdata.TraceData) (droppedSpans int, err error) {
+	tbatch := []*zipkinmodel.SpanModel{}
+	for _, span := range td.Spans {
+		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
+		if err != nil {
+			// return len(td.Spans) - goodSpans, consumererror.Permanent(err)
+			return len(td.Spans), consumererror.Permanent(err)
+		}
+		zs := ze.zipkinSpan(td.Node, sd)
+		tbatch = append(tbatch, &zs)
+	}
+
+	body, err := ze.serializer.Serialize(tbatch)
+	if err != nil {
+		// r.logger.Printf("failed when marshalling the spans batch: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+
+	req, err := http.NewRequest("POST", ze.url, bytes.NewReader(body))
+	if err != nil {
+		// r.logger.Printf("failed when creating the request: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+	req.Header.Set("Content-Type", ze.serializer.ContentType())
+	/*
+		if r.reqCallback != nil {
+			r.reqCallback(req)
+		}*/
+
+	client := &http.Client{Timeout: defaultTimeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		// r.logger.Printf("failed to send the request: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+	_ = resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
```

Comment:
Actually zipkin only replies with 202, does it make sense to be tolerant to other status codes?

Reply:
@huikang Please follow @jcchavezs recommendation.

Reply:
Will do, thanks.

Reply:
@huikang you can change this to `resp.StatusCode != http.StatusAccepted` per previous conversation. In order to not break tests you will need to also do a corresponding change on the test and also on the zipkin receiver test: on `./receiver/zipkinreceiver/trace_receiver_test.go` you will need to add `w.WriteHeader(http.StatusAccepted)` to the handler function used in the test, it will have to look like:
```go
	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		io.Copy(buf, r.Body)
		_ = r.Body.Close()
		w.WriteHeader(http.StatusAccepted)
	}))
```

Reply:
@pjanotti , I remember at some point @jcchavezs mentioned that we could use `if resp.StatusCode < 200 || resp.StatusCode > 299` to make it consistent with the upstream https://github.com/openzipkin/zipkin-go/blob/6c800c64770d0b1e9f07f517c58f58460512fbf4/reporter/http/http.go#L169, 
so I changed to `if resp.StatusCode < 200 || resp.StatusCode > 299`.
If you think we should use `resp.StatusCode != http.StatusAccepted`, please let me know. Thanks.

Reply:
hum, ok Zipkin server always answers 202 but the zipkin reporter itself had the code as we have right now. Let's keep them the behavior of the reporter, ie.: as it is since it was the previous behavior.